host_core/lib/host_core/providers/provider_module.ex

```diff
@@ -395,6 +395,7 @@ defmodule HostCore.Providers.ProviderModule do
           %State{state | healthy: true}
 
         {:ok, _body} ->
+          publish_health_status(state)
```

Comment:
If we're always going to publish status events, I think this block should be refactored. Something like:
```elixir
new_state = case res do
  {:ok, _body} when not state.healthy ->
    %State(state | healthy: true}
  {:error, _} when state.healthy ->
    %State(state | healthy: false}
  _ ->
    state
end
publish_health_status(new_state)
{:noreply, new_state}
```

(`publish_health_status` might need to change too, to infer what the name of the cloud event should be from whether the state is healthy or not)

This separates updating state, which we want to do when the health status changes, from publishing, which we want to do every time now

Reply:
Thanks for the suggestion! I refactored the code and changes `publish_health_status(new_state)` to handle the cases for the passed, failed health event and the normal status event.